### PR TITLE
Spsh 545, PersonPermissions per Inject

### PR DESCRIPTION
**Pull request by kristoff-kiefer**
# Description

Provides a decorator which will inject PersonPermissions into a backend call if desired

## Links to Tickets or other pull requests
https://ticketsystem.dbildungscloud.de/browse/SPSH-545

## Approval for review
- [ ] QA: In addition to review, the code has been manually tested (if manual testing is possible)
- [ ] All points were discussed with the ticket creator, support-team or product owner. The code upholds all quality guidelines from the PR-template.

### CODE REVIEW

**Review by marode-cap — APPROVED** [review]

**Review by pkleybolte — CHANGES_REQUESTED** [review]

File: src/modules/authentication/passport/oidc.strategy.ts [hunk 1]
```diff
@@ -22,21 +23,21 @@ export class OpenIdConnectStrategy extends PassportStrategy(Strategy, 'oidc') {
...
             const refreshToken: string | undefined = tokenset.refresh_token;
 
-            const user: AuthorizationParameters = {
+            const user: AuthorizationParameters & PassportUser = {
                 id_token: idToken,
                 access_token: accessToken,
                 refresh_token: refreshToken,
-                userinfo,
+                userinfo: userinfo,
+                personPermissions: () => Promise.reject(),
```
[the remark this answers is not in the document]
Comment (kristoff-kiefer) [reply src/modules/authentication/passport/oidc.strategy.ts:39]: Weil's zum Type gehört und gefüllt werden sollte

File: src/modules/personenkontext/api/dbiam-personenkontext.controller.integration-spec.ts [hunk 2]
```diff
@@ -74,11 +74,12 @@ describe('dbiam Personenkontext API', () => {
         rolleRepo = module.get(RolleRepo);
 
         await DatabaseTestModule.setupDatabase(orm);
-        app = module.createNestApplication();
+        app = module.createNestApplication().enableShutdownHooks();
```
[the remark this answers is not in the document]
Comment (kristoff-kiefer) [reply src/modules/personenkontext/api/dbiam-personenkontext.controller.integration-spec.ts:77]: No that was an experiment, remnants of which seemed to have stuck

File: src/modules/authentication/api/permissions.decorator.ts [hunk 3]
```diff
@@ -0,0 +1,19 @@
...
+
+export const Permissions: (
+    ...dataOrPipes: (Type<PipeTransform> | PipeTransform<unknown, unknown> | unknown[])[]
+) => ParameterDecorator = createParamDecorator(
+    (_data: unknown, ctx: ExecutionContext): Promise<PersonPermissions | undefined> => {
+        const request: Request = ctx.switchToHttp().getRequest();
+        const passportUser: PassportUser | undefined = request.passportUser;
+
+        if (!passportUser) {
+            return Promise.resolve(undefined);
```
[the remark this answers is not in the document]
Comment (pkleybolte) [reply src/modules/authentication/api/permissions.decorator.ts:14]: Ja, wir haben drüber gesprochen. Meine Erinnerung: Ich habe gesagt, das `undefined` aus TS-Sicht ein valider Rückgabewert ist. Aber da man fachlich nichts mit undefined anfangen kann, weil der Nutzer dann keine Permissions hat, wollen wir direkt `rejecten`.

File: jest.config.cjs [hunk 4]
```diff
@@ -14,6 +14,7 @@ const commonConfig = {
                 useESM: true,
             },
         ],
+        "^.+\\.js$": "<rootDir>/../node_modules/babel-jest"
```
[the remark this answers is not in the document]
Comment (kristoff-kiefer) [reply jest.config.cjs:17]: Make sure that the ts-to-js-mapping comes from a definite place. I hoped it would fix our testing issues

**Review by pkleybolte — APPROVED** [review]

**Pull request merged**